internal/querycoordv2/meta/channel_dist_manager.go

```diff
@@ -360,9 +367,16 @@ func (m *ChannelDistManager) GetShardLeader(channelName string, replica *Replica
 
 				candidateIsStreamingNode := m.checkIfStreamingNode(candidates.Node)
 				channelIsStreamingNode := m.checkIfStreamingNode(channel.Node)
+				logger.Debug("booleans",
+					zap.Bool("candidatesServiceable", candidatesServiceable),
+					zap.Bool("channelServiceable", channelServiceable),
+					zap.Bool("candidateIsStreamingNode", candidateIsStreamingNode),
+					zap.Bool("channelIsStreamingNode", channelIsStreamingNode))
+
 				if channelIsStreamingNode && !candidateIsStreamingNode {
 					// When upgrading from 2.5 to 2.6, the delegator leader may not locate at streaming node.
 					// We always use the streaming node as the delegator leader to avoid the delete data lost when loading segment.
+					logger.Debug("set candidate channel", zap.String("channel", channel.GetChannelName()))
```

Comment:
prefer `set delegator on streamnode to candidate shard leader` and add `zap.String("node", channel.Node)`